File: src/Entity/Offre.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Offre
 *
 * @ORM\Table(name="offre", indexes={@ORM\Index(name="categorieOffre", columns={"CatOffreId"}), @ORM\Index(name="trajetOffre", columns={"IdTrajetOffre"})})
 * @ORM\Entity
 */
class Offre
{
    /**
     * @var int
     *
     * @ORM\Column(name="IdOffre", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idoffre;

    /**
     * @var int
     *
     * @ORM\Column(name="IdCatColis", type="integer", nullable=false)
     */
    private $idcatcolis;

    /**
     * @var string
     *
     * @ORM\Column(name="CatOffreId", type="string", length=255, nullable=false)
     */
    private $catoffreid;

    /**
     * @var string
     *
     * @ORM\Column(name="IdTrajetOffre", type="string", length=255, nullable=false)
     */
    private $idtrajetoffre;

    /**
     * @var string
     *
     * @ORM\Column(name="DescriptionOffre", type="string", length=255, nullable=false)
     */
    private $descriptionoffre;

    /**
     * @var string
     *
     * @ORM\Column(name="MaxRetard", type="string", length=255, nullable=false)
     */
    private $maxretard;

    /**
     * @var float
     *
     * @ORM\Column(name="prixOffre", type="float", precision=10, scale=0, nullable=false)
     */
    private $prixoffre;

    /**
     * @var string|null
     *
     * @ORM\Column(name="DateOffre", type="string", length=255, nullable=true, options={"default"="NULL"})
     */
    private $dateoffre = 'NULL';

    /**
     * @var string|null
     *
     * @ORM\Column(name="DateSortieOffre", type="string", length=255, nullable=true, options={"default"="NULL"})
     */
    private $datesortieoffre = 'NULL';

    /**
     * @var string
     *
     * @ORM\Column(name="Etat", type="string", length=255, nullable=false)
     */
    private $etat;

    /**
     * @var int
     *
     * @ORM\Column(name="nbreDemande", type="integer", nullable=false)
     */
    private $nbredemande;

    /**
     * @var int
     *
     * @ORM\Column(name="IdUser", type="integer", nullable=false)
     */
    private $iduser;

    public function getIdoffre(): ?int
    {
        return $this->idoffre;
    }

    public function getIdcatcolis(): ?int
    {
        return $this->idcatcolis;
    }

    public function setIdcatcolis(int $idcatcolis): self
    {
        $this->idcatcolis = $idcatcolis;

        return $this;
    }

    public function getCatoffreid(): ?string
    {
        return $this->catoffreid;
    }

    public function setCatoffreid(string $catoffreid): self
    {
        $this->catoffreid = $catoffreid;

        return $this;
    }

    public function getIdtrajetoffre(): ?string
    {
        return $this->idtrajetoffre;
    }

    public function setIdtrajetoffre(string $idtrajetoffre): self
    {
        $this->idtrajetoffre = $idtrajetoffre;

        return $this;
    }

    public function getDescriptionoffre(): ?string
    {
        return $this->descriptionoffre;
    }

    public function setDescriptionoffre(string $descriptionoffre): self
    {
        $this->descriptionoffre = $descriptionoffre;

        return $this;
    }

    public function getMaxretard(): ?string
    {
        return $this->maxretard;
    }

    public function setMaxretard(string $maxretard): self
    {
        $this->maxretard = $maxretard;

        return $this;
    }

    public function getPrixoffre(): ?float
    {
        return $this->prixoffre;
    }

    public function setPrixoffre(float $prixoffre): self
    {
        $this->prixoffre = $prixoffre;

        return $this;
    }

    public function getDateoffre(): ?string
    {
        return $this->dateoffre;
    }

    public function setDateoffre(?string $dateoffre): self
    {
        $this->dateoffre = $dateoffre;

        return $this;
    }

    public function getDatesortieoffre(): ?string
    {
        return $this->datesortieoffre;
    }

    public function setDatesortieoffre(?string $datesortieoffre): self
    {
        $this->datesortieoffre = $datesortieoffre;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getNbredemande(): ?int
    {
        return $this->nbredemande;
    }

    public function setNbredemande(int $nbredemande): self
    {
        $this->nbredemande = $nbredemande;

        return $this;
    }

    public function getIduser(): ?int
    {
        return $this->iduser;
    }

    public function setIduser(int $iduser): self
    {
        $this->iduser = $iduser;

        return $this;
    }


}

File: src/com/mycompany/myapp/entities/Offre.java
package com.mycompany.myapp.entities;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


import com.codename1.l10n.SimpleDateFormat;
import java.text.DateFormat;
import java.util.Date;

/**
 *
 * @author utilisateur
 */
public class Offre {
     private  int IdOffre, IdCatColis,IdUser;
         private  int   nbreDemande;
    private int IdTrajetOffre,CatOffreId;   
    private String DescriptionOffre, MaxRetard,Etat;
    private TrajetOffre IdTrajetOffres ;
    private float prixOffre;
    private Date DateOffre;
    private Date DateSortieOffre;

    public Offre() {
    }

    @Override
    public String toString() {
        return "Offre{" + "IdOffre=" + IdOffre + ", IdCatColis=" + IdCatColis + ", IdUser=" + IdUser + ", nbreDemande=" + nbreDemande + ", IdTrajetOffre=" + IdTrajetOffre + ", CatOffreId=" + CatOffreId + ", DescriptionOffre=" + DescriptionOffre + ", MaxRetard=" + MaxRetard + ", Etat=" + Etat + ", IdTrajetOffres=" + IdTrajetOffres + ", prixOffre=" + prixOffre + ", DateOffre=" + DateOffre + ", DateSortieOffre=" + DateSortieOffre + '}';
    }

    public Offre(int IdOffre, int nbreDemande, int IdTrajetOffre, String DescriptionOffre, String MaxRetard, int CatOffreId, float prixOffre, Date DateOffre, Date DateSortieOffre) {
        this.IdOffre = IdOffre;
        this.nbreDemande = nbreDemande;
        this.IdTrajetOffre = IdTrajetOffre;
        this.DescriptionOffre = DescriptionOffre;
        this.MaxRetard = MaxRetard;
        this.CatOffreId = CatOffreId;
        this.prixOffre = prixOffre;
        this.DateOffre = DateOffre;
        this.DateSortieOffre = DateSortieOffre;
    }

    public Offre(int IdTrajetOffre, String DescriptionOffre, String MaxRetard, int CatOffreId, float prixOffre, Date DateOffre, Date DateSortieOffre) {
        this.IdTrajetOffre = IdTrajetOffre;
        this.DescriptionOffre = DescriptionOffre;
        this.MaxRetard = MaxRetard;
        this.CatOffreId = CatOffreId;
        this.prixOffre = prixOffre;
        this.DateOffre = DateOffre;
        this.DateSortieOffre = DateSortieOffre;
    }

    public int getIdOffre() {
        return IdOffre;
    }

    public int getIdCatColis() {
        return IdCatColis;
    }

    public int getIdUser() {
        return IdUser;
    }

    public int getNbreDemande() {
        return nbreDemande;
    }

    public int getIdTrajetOffre() {
        return IdTrajetOffre;
    }

    public String getDescriptionOffre() {
        return DescriptionOffre;
    }

    public String getMaxRetard() {
        return MaxRetard;
    }

    public String getEtat() {
        return Etat;
    }

    public int getCatOffreId() {
        return CatOffreId;
    }

    public TrajetOffre getIdTrajetOffres() {
        return IdTrajetOffres;
    }

    public float getPrixOffre() {
        return prixOffre;
    }

    public Date getDateOffre() {
        return DateOffre;
    }

    public Date getDateSortieOffre() {
        return DateSortieOffre;
    }

    public void setIdOffre(int IdOffre) {
        this.IdOffre = IdOffre;
    }

    public void setIdCatColis(int IdCatColis) {
        this.IdCatColis = IdCatColis;
    }

    public void setIdUser(int IdUser) {
        this.IdUser = IdUser;
    }

    public void setNbreDemande(int nbreDemande) {
        this.nbreDemande = nbreDemande;
    }

    public void setIdTrajetOffre(int IdTrajetOffre) {
        this.IdTrajetOffre = IdTrajetOffre;
    }

    public void setDescriptionOffre(String DescriptionOffre) {
        this.DescriptionOffre = DescriptionOffre;
    }

    public void setMaxRetard(String MaxRetard) {
        this.MaxRetard = MaxRetard;
    }

    public void setEtat(String Etat) {
        this.Etat = Etat;
    }

    public void setCatOffreId(int CatOffreId) {
        this.CatOffreId = CatOffreId;
    }

    public void setIdTrajetOffres(TrajetOffre IdTrajetOffres) {
        this.IdTrajetOffres = IdTrajetOffres;
    }

    public void setPrixOffre(float prixOffre) {
        this.prixOffre = prixOffre;
    }

    public void setDateOffre(Date DateOffre) {
        this.DateOffre = DateOffre;
    }

    public void setDateSortieOffre(Date DateSortieOffre) {
        this.DateSortieOffre = DateSortieOffre;
    }
    
}
